File: EchoBot.java
// EchoBot.java
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.*;

public class EchoBot {
    private List<String> history = new ArrayList<>();
    private List<String> greetings = Arrays.asList("Hello there! 👋", "Hi! 😊", "Greetings, human!", "Yo! 🤖", "Howdy!");
    private Random rand = new Random();

    private void addHistory(String msg) {
        history.add(msg);
        if (history.size() > 20) history.remove(0);
    }

    public String process(String text) {
        addHistory(text);
        if (text.startsWith("/")) {
            String[] parts = text.split(" ", 2);
            String cmd = parts[0].toLowerCase();
            String arg = parts.length > 1 ? parts[1] : "";
            switch (cmd) {
                case "/help": return help();
                case "/greet": return greetings.get(rand.nextInt(greetings.size()));
                case "/time": return LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
                case "/history": return showHistory();
                case "/flip":
                    if (arg.isEmpty()) return "Flip what? Type /flip <message>";
                    return swapCase(arg);
                case "/bye": return "Goodbye! 👋";
                default: return "Unknown command: " + cmd + ". Type /help for commands.";
            }
        }
        return "You said: \"" + text + "\"";
    }

    private String help() {
        return "Available commands:\n" +
               "  /help      - Show this help\n" +
               "  /greet     - Get a random greeting\n" +
               "  /time      - Show current date/time\n" +
               "  /history   - Show recent messages\n" +
               "  /flip <msg> - Echo with swapped case\n" +
               "  /bye       - Exit the bot";
    }

    private String showHistory() {
        if (history.isEmpty()) return "No messages yet.";
        StringBuilder sb = new StringBuilder("Recent messages:\n");
        for (int i = 0; i < history.size(); i++) {
            sb.append("  [").append(i+1).append("] ").append(history.get(i)).append("\n");
        }
        return sb.toString().trim();
    }

    private String swapCase(String s) {
        char[] arr = s.toCharArray();
        for (int i = 0; i < arr.length; i++) {
            if (Character.isLowerCase(arr[i])) arr[i] = Character.toUpperCase(arr[i]);
            else if (Character.isUpperCase(arr[i])) arr[i] = Character.toLowerCase(arr[i]);
        }
        return new String(arr);
    }

    public static void main(String[] args) {
        EchoBot bot = new EchoBot();
        Scanner scanner = new Scanner(System.in);
        System.out.println("🤖 Echo Bot: Hello! Type something or /help.");
        while (true) {
            System.out.print("You: ");
            String user = scanner.nextLine().trim();
            if (user.isEmpty()) continue;
            String response = bot.process(user);
            if (user.contains("/bye")) {
                System.out.println("🤖 Echo Bot: " + response);
                break;
            }
            System.out.println("🤖 Echo Bot: " + response);
        }
        scanner.close();
    }
}

File: echo_bot.js
// echo_bot.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

class EchoBot {
    constructor() {
        this.history = [];
        this.greetings = ["Hello there! 👋", "Hi! 😊", "Greetings, human!", "Yo! 🤖", "Howdy!"];
    }

    addHistory(msg) {
        this.history.push(msg);
        if (this.history.length > 20) this.history.shift();
    }

    process(text) {
        this.addHistory(text);
        if (text.startsWith('/')) {
            const parts = text.split(' ');
            const cmd = parts[0].toLowerCase();
            const arg = parts.slice(1).join(' ');
            switch (cmd) {
                case '/help': return this.help();
                case '/greet': return this.greetings[Math.floor(Math.random() * this.greetings.length)];
                case '/time': return new Date().toLocaleString();
                case '/history': return this.showHistory();
                case '/flip':
                    if (!arg) return "Flip what? Type /flip <message>";
                    return arg.split('').map(c => c === c.toUpperCase() ? c.toLowerCase() : c.toUpperCase()).join('');
                case '/bye': return "Goodbye! 👋";
                default: return `Unknown command: ${cmd}. Type /help for commands.`;
            }
        }
        return `You said: "${text}"`;
    }

    help() {
        return `Available commands:
  /help      - Show this help
  /greet     - Get a random greeting
  /time      - Show current date/time
  /history   - Show recent messages
  /flip <msg> - Echo with swapped case
  /bye       - Exit the bot`;
    }

    showHistory() {
        if (this.history.length === 0) return "No messages yet.";
        const lines = this.history.map((msg, i) => `  [${i+1}] ${msg}`);
        return "Recent messages:\n" + lines.join('\n');
    }
}

function ask(question) {
    return new Promise(resolve => rl.question(question, resolve));
}

async function main() {
    const bot = new EchoBot();
    console.log("🤖 Echo Bot: Hello! Type something or /help.");
    while (true) {
        const user = await ask("You: ");
        if (!user.trim()) continue;
        const response = bot.process(user);
        if (user.includes('/bye')) {
            console.log(`🤖 Echo Bot: ${response}`);
            break;
        }
        console.log(`🤖 Echo Bot: ${response}`);
    }
    rl.close();
}

main().catch(console.error);
